lib/systems/examples.nix

```diff
@@ -204,4 +204,10 @@ rec {
     libc = "msvcrt"; # This distinguishes the mingw (non posix) toolchain
     platform = {};
   };
+
+  # Ghcjs
+  ghcjs = {
+    config = "js-unknown-none-ghcjs";
```

Comment:
I'm not sure about this. While I agree that we need better cross-compilation support for ghcjs, "js-unknown-none-ghcjs" is not a widely agreed upon name. [Searching for js-unknown-none-ghcjs](https://www.google.com/search?q=js-unknown-none-ghcjs) only gives me #65589. I was going to recommend using "asmjs-unknown-emscripten" which is used by Rust as well, but then realized that GHCJS is not technically asmjs. In addition, we have no C compiler that is compatible with GHCJS, so things like bootstrapping are not well defined.

Reply:
I think it's okay to allow this one though. GHCJS is in a pretty unique situation, so there is not that much downside to allowing it on an experimental basis. I would like to get some agreement with @luite and ghcjs maintainers on this though.

Reply:
I would suggest using ```js-unknown-ghcjs``` though. ```none-ghcjs``` doesn't really make sense. The "environment" is supposed to qualify the "os", but that doesn't make sense when your os is none. For instance try this:

```
$ ./config.sub asmjs-unknown-none-ghcjs
Invalid configuration `asmjs-unknown-none-ghcjs': system `none-ghcjs' not recognized
```

Reply:
@matthewbauer thanks for the review. I agree that `js-unknown-ghcjs` is better. The primary reason for choosing that is that it's already pretty close to how cabal thinks about ghcjs.

We (that includes @luite) will move ghcjs more towards a proper ghc cross compiler. @Ericson2314 will be able to supply you with some more background knowledge from ICFP I guess.

@luite *is* working on some emscripten support for ghcjs, so we may eventually interact with emscripten, however that may only work from within cabal anyway due to calling convention needs and as such never really be the same as asmjs targets. If we can make that work eventually we may need to add another target.

Note that this needs to be fixed in Nixpkgs, as we can not locally fix it via overlays, as per #65589.

input-output-hk/haskell.nix's overlay branch (input-output-hk/haskell.nix#210) will allow you to treat ghcjs like a regular cross compiler as in
```
nix build '(with import ./. { nixpkgs = ../nixpkgs; nixpkgsArgs = { crossSystem = { config = "js-unknown-ghcjs"; }; }; }; (haskell-nix.hackage-package { name = "hello"; version = "1.0.0.2"; })).components.exes.hello'
```
after removing a few more minor kinks.